app/controllers/media_objects_controller.rb

```diff
@@ -95,8 +95,13 @@ def custom_update
     flash[:notice] = @notice
   end
 
+  def index
+    respond_to do |format|
+      format.json   { render json: MediaObject.all.map{|mo| mo.to_json } }
```

Comment:
I kind of like this approach: http://metabates.com/2012/02/22/adding-pagination-to-an-api/ Or is there another way you had in mind?